### PR TITLE
[structured config] Migrate resources from project-fully-featured to struct config resources

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Migrates the experimental copy of `project-fully-featured` to use struct config resources instead of traditional function-based resources.

## Test Plan

Existing unit tests.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11773** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11773" target="_blank">Graphite</a> 
    * **PR #11645** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11645" target="_blank">Graphite</a> 
      * **PR #11785** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11785" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11785?utm_source=stack-comment).

File: examples/experimental/project_fully_featured_v2_resources/project_fully_featured_v2_resources/resources/common_bucket_s3_pickle_io_manager.py [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+class CommonBucketS3PickleIOManager(StructuredConfigIOManagerBase):
     """
     A version of the s3_pickle_io_manager that gets its bucket from another resource.
     """
-    return s3_pickle_io_manager(
-        build_init_resource_context(
-            config={"s3_bucket": init_context.resources.s3_bucket},
-            resources={"s3": init_context.resources.s3},
+
+    s3_bucket: ResourceDependency[str]
```
Comment (schrockn) [inline examples/experimental/project_fully_featured_v2_resources/project_fully_featured_v2_resources/resources/common_bucket_s3_pickle_io_manager.py:14]: does this even need to be a resource?

**Review by schrockn — CHANGES_REQUESTED** [review]
I think this feels pretty good overall. Can we make s3_bucket a bare string?

File: examples/experimental/project_fully_featured_v2_resources/project_fully_featured_v2_resources/resources/common_bucket_s3_pickle_io_manager.py (hunk 1 above)
Comment (benpankow) [reply examples/experimental/project_fully_featured_v2_resources/project_fully_featured_v2_resources/resources/common_bucket_s3_pickle_io_manager.py:14]: no, good point

**Review by schrockn — CHANGES_REQUESTED** [review]
Looks like a rebase error

File: examples/experimental/project_fully_featured_v2_resources/project_fully_featured_v2_resources/resources/hn_resource.py [hunk 2]
```diff
@@ -39,19 +41,20 @@ def min_item_id(self) -> int:
 
 
 class HNSnapshotClient(HNClient):
-    def __init__(self):
+    @cached_method
```
Comment (schrockn) [inline examples/experimental/project_fully_featured_v2_resources/project_fully_featured_v2_resources/resources/hn_resource.py:44]: Seems a bit odd to auto-cache this

**Review by schrockn — APPROVED** [review]
This is great

**Pull request merged**